src/hooks/useSatelliteManifests.ts

```diff
@@ -21,32 +21,44 @@ interface SatelliteManifestAPIData {
   };
 }
 
-const fetchSatelliteManifestData = (): Promise<any> => {
+const fetchSatelliteManifestData = async (
+  offset = 0,
+  previousManifests: ManifestEntry[] = []
+): Promise<ManifestEntry[]> => {
   const jwtToken = Cookies.get('cs_jwt');
   const { rhsmAPIBase } = getConfig();
-  return fetch(`${rhsmAPIBase}/management/v1/allocations`, {
-    headers: { Authorization: `Bearer ${jwtToken}` },
-    mode: 'cors'
-  }).then((response) => response.json());
-};
 
-const getOnlySatelliteManifests = (data: ManifestEntry[]): ManifestEntry[] => {
-  return data.filter((manifest: ManifestEntry) => manifest.type === 'Satellite');
+  const response = await fetch(
+    `${rhsmAPIBase}/management/v1/allocations?type=Satellite&offset=${offset}`,
+    {
+      headers: { Authorization: `Bearer ${jwtToken}` },
+      mode: 'cors'
+    }
+  );
+
+  const manifestResponseData: SatelliteManifestAPIData = await response.json();
+
+  const { count, limit } = manifestResponseData.pagination;
+
+  const combinedManifests = [...previousManifests, ...manifestResponseData.body];
+
+  if (count === limit) {
+    // Fetch the next page's data
+    const newOffset = offset + limit;
+
+    return await fetchSatelliteManifestData(newOffset, combinedManifests);
+  } else {
+    return combinedManifests;
+  }
```

Comment:
It feels a little awkward for `fetchSatelliteManifestData` to accept a random array of manifests as an arg.  The inner call has no reason to know about what happened in the outer call.  Could we just move where the manifests are combined, something like this:

```ts
const manifestResponseData: SatelliteManifestAPIData = await response.json();

const { count, limit } = manifestResponseData.pagination;

const manifests = manifestResponseData.body;

if (count === limit) {
  const additionalManifests = await fetchSatelliteManifestData(newOffset);
  manifests = [...manifests, ...additionalManifests];
}

return manifests;
```

This would also reduce you down to just one return statement which is nice.

Reply:
ok just updated.